api/docker-compose.yaml

```diff
@@ -20,165 +26,52 @@ services:
     ports:
       - 127.0.0.1:3306:3306
   api:
+    <<: *api-defaults
     depends_on:
       - db
-    build:
-      context: ./src/dev/server
-    image: allofustest/workbench-dev-api:local
-    user: ${UID}
-    working_dir: /w/api
-    volumes:
-      - src-sync:/w:nocopy
-      - gradle-cache:/.gradle
-      - ~/.config:/.config:cached
-      - ~/.gsutil:/.gsutil:cached
     command: ./project.rb start-api-and-incremental-build
-    environment:
-      - GOOGLE_APPLICATION_CREDENTIALS=/w/api/sa-key.json
-    env_file:
-      - db/vars.env
     ports:
       - 127.0.0.1:8081:8081
       - 127.0.0.1:8001:8001
   es-scripts:
-    build:
-      context: ./src/dev/server
-    image: allofustest/workbench-dev-api:local
-    user: ${UID}
-    working_dir: /w/api
-    environment:
-      - GOOGLE_APPLICATION_CREDENTIALS=/w/api/sa-key.json
-    volumes:
-      - src-sync:/w:nocopy
-      - gradle-cache:/.gradle
-      - ~/.config:/.config:cached
-      - ~/.gsutil:/.gsutil:cached
+    <<: *api-defaults
   db-scripts:
+    <<: *api-defaults
     depends_on:
       - db
-    build:
-      context: ./src/dev/server
-    image: allofustest/workbench-dev-api:local
-    user: ${UID}
     working_dir: /w/api/db
-    volumes:
-      - src-sync:/w:nocopy
-      - gradle-cache:/.gradle
     entrypoint: ['with-uid.sh', 'wait-for-it', 'db:3306', '-t', '30', --]
```

Comment:
Maybe parameterize the port number and timeout?

Reply:
See above comment. Also the port is fine, but the timeout is a constant. Creating knobs for things that don't need to change generally just adds noise. If the intent was to avoid duplication with the other services here, see the PR description: my preference is to just eliminate the services with duplicate functionality. If there's a legitimate need for multiple services with this entrypoint, using an anchor can also work to avoid double-encoding of this.